### PR TITLE
Build-a-Stand Mk1: JoJo Harder

### PR DESCRIPTION
**Pull request by ghost**
# HERE WE GO BOIS, THE HOLOPARASITE REWORK WE'VE ALL BEEN WAITING FORRR!

Based off my work at https://github.com/HippieStation/HippieStation/pull/11848, but this time ported by somebody competent instead of Toxic.

Any guardian can have 1 major ability and multiple minor abilities.

## PR to-do

- [x] Figure out how to make the TGUI look right.
- [ ] ~~Re-add toy holoparasites~~
- [x] Add the stand arrow.
- [ ] Maybe add an actual ruin for the stand arrow? If I can figure out how to make a HARD ruin.

## Major Guardian Abilities

**Assassin**: The holopara can enter "stealth mode" -- when it attacks someone in stealth mode, it'll penetrate armor and deal 50+ damage. Cooldown for stealth mode is affected by potential.

**Explosive**: The holopara can alt-click on any object to turn it into a bomb temporarily. Bumping into a dense/solid bomb, or clicking on it, will blow up the person who did it, usually KOing and possibly dismembering them. How long a bomb stays a bomb is affected by potential.

**Frenzy**: The holopara makes it's summoner move fast while manifested, and it can click on people within it's range to rush them, and it also has an ability to teleport behind people and attack them (omae wa shinderu. nani?!). Potential has no affect. (Range C or + only)

**Gravitokinetic**: The holopara can high gravity to tiles and people, which slows them down and deals brute damage over time. How far you can get from someone before the gravity is lifted is affected by potential.

**The Hand**: The holopara can click on any tile at range to bring all it's unanchored contents in front of you. People, lockers, medkits, et cetera. It has a cooldown, affected by potential.

**Healing**: The holopara can click on any mob or object to heal it. Healing amount is based on potential.

**Predator**: The holopara can scan blood DNA and fingerprints, allowing it to track anyone who it has scanned. Tracking precision is based on potential.

**Scout**: The holopara can go into "scout mode", which turns it near-invisible and gives it infinite range, but it cannot attack anyone in this state.

**Time Erasure**: The holopara can "skip time" -- this spawns clones of the holopara and it's summoner, turning the originals invisible and invincible for the duration. They however, cannot attack anyone during this period. Duration is affected by potential.

## Minor (secondary) Guardian Abilities

**Surveillance Snares**: gives you a alert if any mob (including yourself) walks over 

**Teleportation Pad**: Basically the exact same thing as the Traitor one.

## Requiem Guardian Abilities (unobtainable through normal means for now)

**Dimensional Manifestation**: The holopara can summon the (very obvious) manifestation of a 6x6 pocket dimension into the world on top of it. If you are inside the manifestation when the guardian de-manifests it, you and anything you are pulling (neckgrab for mobs) will be teleported into it. The guardian can move the pocket dimension around like a camera while inside, and re-manifest it whereever they wish. There is no normal way for anyone but the guardian and it's master to escape. It has air, gravity, and power, but no telecommunications.

**Absolution**: While your holopara is manifested, you are immune to nearly all attacks. Tasers and disablers will return to sender, people will stunbaton themselves instead of you, smoke and foam will clear around you, no bad air will affect you, explosions will simply go around you. The holopara, however, is still vulnerable to attack, and it will still transfer damage to you.

**Timestop**: You Stop Time on a 3x3 area around you, you and your holopara are not affected by this, however, everything else is, projectiles, people, explosions, you name it.

:cl:
add: Guardians/holoparasites are no longer seperated into types
add: Instead, you get to "build" one, with a limited amount of points, selecting different stats and abilities
add: The following major abilities are available to holoparasites: Assassin, Explosive, Frenzy, Gravitokinetic, The Hand, Healing, Predator, Scout, and Time Erasure.
add: The following minor abilities are available to holoparasites: Surveillance Snares, Teleportation Pad
del: Holoparasite types that didn't make the cut: Toy, Lightning, Protector, Dextrous, Charger, Chaos
/:cl:

### CODE REVIEW

**Comment by Coolsurf6** [comment]
This has some real "potential". Looking forward to it lad!

**Comment by ghost** [comment]
UI demo
bqoEVZgAR5

**Comment by ghost** [comment]
Dimensional Manifestation demo

**Comment by PowerfulBacon** [comment]
Very cool

**Comment by derpymcfuckwit** [comment]
agh im coooming!! yes

**Comment by TeomanTheGreat** [comment]
Kek

**Comment by Coolsurf6** [comment]
Hold on. Who actually makes the stand though? Is it the person who uses the cards, then creates it, or does the ghost who gets chosen to be the stand choose their powers?

**Comment by ghost** [comment]
@coolsurf6 user of the cards/injector

**Comment by Coolsurf6** [comment]
Does it affect the cards that randomly give, like the ones found on lavaland? How do they work or are all cards and injectors now no longer random? I'm a bit confused on how it works but it looks really cool.

**Comment by ghost** [comment]
only thing that is random now are stand arrows

**Comment by Coolsurf6** [comment]
> only thing that is random now are stand arrows

Would it be safe to change the card spawns that are on lavaland with the stand arrow now?

**Comment by ghost** [comment]
stand arrow has a chance to dust you though

**Comment by TeomanTheGreat** [comment]
@steamp0rt instead of dust why doesn't it just have the same effect as suicide? (Like in JoJo golden wind when that janitor guy in the beginning got stabbed by black sabbath's stand arrow and his soul went yeet and he died with no physical damage on his body)

Edit: https://youtu.be/oMJp1tTcQn8

**Comment by ghost** [comment]
so there's absolutely no edge cases of possible ways to revive

**Comment by TeomanTheGreat** [comment]
https://youtu.be/oMJp1tTcQn8

**Comment by Coolsurf6** [comment]
Yeah it is a bit weirder. Maybe just disconnect the soul from the body, so it can't be revived. Like how ghosting works or the do not resurrect button.

Dusting is a good alternative though.

File: code/game/machinery/cloning.dm [hunk 1]
```diff
@@ -181,27 +181,36 @@
 	if(!empty) //Doesn't matter if we're just making a copy
 		clonemind = locate(mindref) in SSticker.minds
 		if(!istype(clonemind))	//not a mind
+			to_chat(world, "1")
```
Comment (qwertyquerty) [inline code/game/machinery/cloning.dm]: Remove debug code

**Comment by TeomanTheGreat** [comment]
For the stand arrow ruin make it 2 goliath broodmothers guarding it on lavaland lmao

**Comment by qwertyquerty** [comment]
Further checkboxes will be done in separate PR

**Pull request merged**